import { Request, Response, NextFunction } from 'express';
const handleErrorAsync =
  (fn: (req: Request, res: Response, next: NextFunction) => Promise<void>) =>
  (req: Request, res: Response, next: NextFunction) => {
    fn(req, res, next).catch(next);
  };

export default handleErrorAsync;
